### PR TITLE
fix: organize README.md build instructions

### PR DESCRIPTION
**Pull request by matthewpereira**
Spread out instructions for building OpenFGA and number them sequentially.

## Description
Previously, some instructions were grouped and others separated into multiple code blocks. Additionally, two different workflows were presented under a single heading.

This change attempts to make the build instructions easier to navigate by offering smaller chunks of instructions, and numbering them within clearly labelled sections.

Before comparison:
image

After comparison:
image

## Review Checklist
- [ ] I have added documentation for new/changed functionality in this PR or in [openfga.dev](https://github.com/openfga/openfga.dev)
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,30 +58,55 @@ with the command:
 ./bin/openfga run
 ```
 
-### Source
+### Building from Source
+
+There are two recommended options for building from OpenFGA source code:
```
Comment (jon-whit) [inline README.md:63]: typo, should be:


> for building OpenFGA from source code:

Comment (jon-whit) [inline README.md:63]: typo, should be:


> for building OpenFGA from source code:

File: README.md [hunk 2]
```diff
@@ -3,11 +3,11 @@
 [![Go Reference](https://pkg.go.dev/badge/github.com/openfga/openfga.svg)](https://pkg.go.dev/github.com/openfga/openfga)
 ![GitHub release (latest SemVer)](https://img.shields.io/github/v/release/openfga/openfga?sort=semver&color=green)
 [![Container Image](https://img.shields.io/github/v/release/openfga/openfga?color=blueviolet&label=container&logo=docker "Container Image")](https://hub.docker.com/r/openfga/openfga/tags)
-![Downloads](https://img.shields.io/github/downloads/openfga/openfga/total.svg?style=flat&color=lightgrey)  <!-- render new line with double space here -->  
+![Downloads](https://img.shields.io/github/downloads/openfga/openfga/total.svg?style=flat&color=lightgrey) <!-- render new line with double space here -->
```
Comment (craigpastro) [inline README.md:6]: I think the double space at the end is intentional to render the badges on a newline. Do we want to keep doing that or just render them on one line?

File: README.md [hunk 3]
```diff
@@ -57,30 +58,55 @@ with the command:
...
 
-```bash
-git clone https://github.com/openfga/openfga.git && cd openfga
-go build cmd/openfga/openfga.go
+3. Run the server with:
 
-./openfga run
-```
+   ```bash
+   ./openfga
```
Comment (craigpastro) [inline README.md:86]: We need to add a `run` command to the binary now:
```suggestion
   ./openfga run
```

File: README.md [hunk 4]
```diff
@@ -57,30 +58,55 @@ with the command:
...
+2. Then use the build command:
+
+   ```bash
+   go build cmd/openfga/openfga.go
+   ```
+
+3. Run the server with:
+
+   ```bash
+   ./openfga
```
Comment (craigpastro) [inline README.md:108]:
```suggestion
   ./openfga run
```

File: README.md [hunk 5]
```diff
@@ -57,30 +58,55 @@ with the command:
 ./bin/openfga run
 ```
 
-### Source
+### Building from Source
+
+There are two recommended options for building from OpenFGA source code:
+
+#### a. Building from source with `go install`
```
Comment (craigpastro) [inline README.md:65]: In the end we show how to run the binary. Should this be "Running from source with `go install`"?

File: README.md (hunk 2 above)
Comment (matthewpereira) [reply README.md:6]: I think one of our linters removed this - I didn't mean to make a change here. Maybe we need a prettier exception here if we want to keep the double spaces?

File: README.md (hunk 5 above)
Comment (matthewpereira) [reply README.md:65]: 🤷

@jon-whit @rhamzeh @miparnisari do you have any thoughts on this?

Comment (elbuo8) [reply README.md:65]: I'd kinda prefer if we only had one recommended way. I'd vote for `go build`. It's a bit less portable as it doesn't rely on Go details as GOBIN etc.

Comment (matthewpereira) [reply README.md:65]: Maybe the conversation about preferred methods or primary methods or wording about those methods needs to be a separate discussion/RFC. For now, I think we need to get working instructions into the README.md.

**Review by jon-whit — APPROVED** [review]

**Review by jon-whit — DISMISSED** [review]

**Pull request merged**